### PR TITLE
[Serve][Doc] Add http endpoint for dag pattern doc

### PR DESCRIPTION
**Pull request by sihanwang41**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/serve/deployment-graph/http_endpoint_for_dag_graph.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
++++
+
+```{eval-rst}
+.. literalinclude:: ../doc_code/deployment_graph_dag_http.py
+   :language: python
+```
+
+````{note}
+1. Serve provide a special driver (DAGDriver) to accept the http request and drive the dag graph execution
```
Comment (simon-mo) [inline doc/source/serve/deployment-graph/http_endpoint_for_dag_graph.md:15]: can you also provide link to the DAGDriver part of tutorial?

**Review by simon-mo — APPROVED** [review]

File: doc/source/serve/deployment-graph/http_endpoint_for_dag_graph.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Pattern: Http endpoint for dag graph
+
+The example shows how to add endpoint for dag graph
```
Comment (jiaodong) [inline doc/source/serve/deployment-graph/http_endpoint_for_dag_graph.md:3]: How about something like 

"This example shows how to configure ingress component of the deployment graph, such as HTTP endpoint prefix, HTTP to python object input adapter and number of ingress replicas"

File: doc/source/serve/deployment-graph/http_endpoint_for_dag_graph.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+## Code
+
++++
+
+```{eval-rst}
+.. literalinclude:: ../doc_code/deployment_graph_dag_http.py
+   :language: python
+```
+
+````{note}
```
Comment (jiaodong) [inline doc/source/serve/deployment-graph/http_endpoint_for_dag_graph.md:14]: hmm i think this should be three ` like ``` ?

Comment (sihanwang41) [reply doc/source/serve/deployment-graph/http_endpoint_for_dag_graph.md:14]: if you have nested, ```` is the way to indicate this is the outter section. (lmk if you know a better solution)

Comment (jiaodong) [reply doc/source/serve/deployment-graph/http_endpoint_for_dag_graph.md:14]: oh you're nesting them, let me check again on the built html once your latest commit finishes

**Review by jiaodong — APPROVED** [review]
content looks good to me, let's give it a final pass once CI finishes building and see the full version on our website

**Pull request merged**